### DISCUSSION TITLE
Change FutureProvider from error state to loading state

### DISCUSSION THREAD

**Discussion by andreadaoud** [rrousselGit/riverpod#1746] [Q&A]
Hi, I am using `FutureProvider` to load data. When an error occurred inside FutureProvider, I display a retry button for the user. When the user click that button, I want to reset the state of the future provider from error to loading, in order to hide the error message from user. Currently the error state will persist until new data is loaded or new error happened. What can I do? Thanks a lot!

**Comment by hamza-imran75** [comment]
Can you try something like this:
```
if(!state.isLoading && state.hasError) {
... show error
}
```

**Comment by rrousselGit** [comment]
I assume that you're using ref.refresh or ref.invalidate to recompute the future? 

In that scenario, Riverpod voluntarily keeps the previous state as that's commonly what's needed when doing a pull-to-refresh.
But you can determine if the future is recomputing by doing:

```dart
if (ref.watch(myFutureProvider).isRefreshing)
```
